azurekeyvault-flexvolume/oauth.go

```diff
@@ -191,6 +193,39 @@ func GetServicePrincipalToken(tenantId string, env *azure.Environment, resource
 	return nil, fmt.Errorf("no credentials provided for AAD application %s", aADClientID)
 }
 
+func retryFetchToken(req *http.Request, maxAttempts int) (resp *http.Response, err error) {
+	attempt := 0
+	delay := time.Duration(0)
+	client := &http.Client{}
+	const maxDelay time.Duration = 60 * time.Second
+
+	// retry guidance as suggested here - https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance
+	for attempt < maxAttempts {
```

Comment:
How does this correlate with kubelet time out ? Due to the exponential retry scheme, its not clear what is the max wait time that can be incurrent. Can you please add a comment here on the same.

Reply:
@kkmsft Sure, that's what I need to test out and check.

Reply:
these timeout consts might be useful: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/volume_manager.go#L65-L78